internal/jobworker/cgroup/controller.go

```diff
@@ -0,0 +1,215 @@
+package cgroup
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"path"
+	"path/filepath"
+	"strconv"
+
+	"github.com/pkg/errors"
+	"golang.org/x/sys/unix"
+)
+
+// controller enables and applies cgroup controls
+type controller interface {
```

Comment:
nit: push the interface closer to where it is used, not where the types are declared.

Reply:
Completely agree, thanks!

Reply:
Updated